lib/modules/plugin/DeviceManagerEngine.ts

```diff
@@ -134,28 +142,24 @@ export class DeviceManagerEngine extends AbstractEngine<DeviceManagerPlugin> {
   }
 
   async onDelete(index: string) {
-    const promises = [];
+    const collections = [
+      InternalCollection.ASSETS,
+      InternalCollection.ASSETS_HISTORY,
+      InternalCollection.ASSETS_GROUPS,
+      InternalCollection.DEVICES,
+      InternalCollection.MEASURES,
+    ];
 
-    promises.push(this.sdk.collection.delete(index, InternalCollection.ASSETS));
-    promises.push(
-      this.sdk.collection.delete(index, InternalCollection.ASSETS_HISTORY)
-    );
-    promises.push(
-      this.sdk.collection.delete(index, InternalCollection.DEVICES)
-    );
-    promises.push(
-      this.sdk.collection.delete(index, InternalCollection.MEASURES)
+    await Promise.all(
+      collections.map(async (collection) => {
+        if (await this.sdk.collection.exists(index, collection)) {
+          await this.sdk.collection.delete(index, collection);
+        }
+      })
```

Comment:
Maybe you can do mExists then mDelete on all existing collections

Reply:
Currently, we can't because `mExists` and `mDelete` doesn't exist on collection controller